tracker-smoothtrack/ftnoir_tracker_smoothtrack.cpp

```diff
@@ -0,0 +1,184 @@
+/* Copyright (c) 2025
+ *
+ * Permission to use, copy, modify, and/or distribute this
+ * software for any purpose with or without fee is hereby granted,
+ * provided that the above copyright notice and this permission
+ * notice appear in all copies.
+ */
+
+#include "ftnoir_tracker_smoothtrack.h"
+#include "api/plugin-api.hpp"
+
+#include <QSocketNotifier>
+#include <cmath>
+#include <iterator>
+
+smoothtrack::smoothtrack() : last_recv_pose{ 0, 0, 0, 0, 0, 0 }, last_recv_pose2{ 0, 0, 0, 0, 0, 0 }
+{
+}
+
+smoothtrack::~smoothtrack()
+{
+    requestInterruption();
+    wait();
+    disconnect_from_device();
+}
+
+bool smoothtrack::connect_to_device()
+{
+    // Try to connect to iOS device via usbmuxd
+    usbmuxd_device_info_t* device_list = nullptr;
+    int device_count = usbmuxd_get_device_list(&device_list);
+
+    if (device_count < 1)
+    {
+        if (device_list)
+            usbmuxd_device_list_free(&device_list);
+        return false;
+    }
+
+    // Use the first available device
+    uint32_t device_handle = device_list[0].handle;
+    usbmuxd_device_list_free(&device_list);
+
+    // Connect to the SmoothTrack port on the iOS device
+    usbmuxd_fd = usbmuxd_connect(device_handle, static_cast<uint16_t>(s.port));
+
+    if (usbmuxd_fd < 0)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+void smoothtrack::disconnect_from_device()
+{
+    if (sock)
+    {
+        sock->close();
+        sock->deleteLater();
+        sock = nullptr;
+    }
+
+    if (usbmuxd_fd >= 0)
+    {
+        close(usbmuxd_fd);
+        usbmuxd_fd = -1;
+    }
+}
+
+void smoothtrack::run()
+{
+    // Connect to the iOS device
+    if (!connect_to_device())
+    {
+        return;
+    }
+
+    // Wrap the native socket in a QTcpSocket
+    sock = new QTcpSocket();
+    if (!sock->setSocketDescriptor(usbmuxd_fd))
+    {
+        disconnect_from_device();
+        return;
+    }
+
+    QByteArray buffer;
+    buffer.resize(sizeof(last_recv_pose));
+
+    while (!isInterruptionRequested())
+    {
+        // Wait for data to be available
+        if (!sock->waitForReadyRead(100))
+        {
+            // Check if we're still connected
+            if (sock->state() != QAbstractSocket::ConnectedState)
+            {
```

Comment:
Consider logging the failure reason with `qDebug() << ...` in the places where the tracker fails.